src/main/java/org/osc/manager/rest/server/api/DeviceApis.java

```diff
@@ -0,0 +1,391 @@
+            @PathParam("memberId") Long memberId, DeviceMemberEntity entity) {
+
+        logger.info("Updating for Device Member with Device Member Id...:" + memberId);
+        logger.info("and Device Id...:" + deviceId);
+
+        return this.txControl.required(new Callable<DeviceMemberEntity>() {
+
+            @Override
+            public DeviceMemberEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = DeviceApis.this.em.getCriteriaBuilder();
+
+                CriteriaQuery<DeviceMemberEntity> query = criteriaBuilder.createQuery(DeviceMemberEntity.class);
+                Root<DeviceMemberEntity> r = query.from(DeviceMemberEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("parent").get("id"), deviceId)),
+                        criteriaBuilder.equal(r.get("id"), memberId));
+
+                List<DeviceMemberEntity> result = DeviceApis.this.em.createQuery(query).getResultList();
+                if (result.isEmpty()) {
+                    throw new Exception("Device or the Device Member Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                result.get(0).setName(entity.getName());
+                DeviceApis.this.em.persist(result.get(0));
+                return result.get(0);
+            }
+        });
+    }
+
+    /**
+     * Deletes the Device Member for a given device Id and the member Id
+     *
+     *
+     */
+    @Path("/{deviceId}/members/{memberId}")
+    @DELETE
+    public void deleteDeviceMember(@PathParam("deviceId") Long deviceId, @PathParam("memberId") Long memberId,
+            DeviceMemberEntity entity) {
+
+        logger.info("Deleting the Device Member with device member Id...:" + memberId);
+
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DeviceEntity result = DeviceApis.this.em.find(DeviceEntity.class, deviceId);
+                if (result == null) {
+                    return null;
+                }
+                DeviceMemberEntity memberResult = DeviceApis.this.em.find(DeviceMemberEntity.class, memberId);
+                if (memberResult == null) {
+                    return null;
+                }
+                DeviceApis.this.em.remove(memberResult);
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Lists the Device Member Id's for a given Device Id
+     *
+     * @return - Device Member Id's
+     * @throws Exception
+     */
+    @Path("/{deviceId}/members")
+    @GET
+    public List<String> getDeviceMembersIds(@PathParam("deviceId") Long deviceId) throws Exception {
+
+        logger.info("Listing Device Members Ids'for device Id ...:" + deviceId);
+
+        return this.txControl.supports(new Callable<List<String>>() {
```

Comment:
Instead this can call the existing listDeviceMembers and extract the ids from it right? This will once again reduce this code.